apm-application-toolkit/apm-toolkit-trace/src/main/java/org/apache/skywalking/apm/toolkit/trace/CallableWrapper.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.skywalking.apm.toolkit.trace;
+
+import java.util.concurrent.Callable;
+
+/**
+ * @author carlvine500
+ */
+@TraceCrossThread
+public class CallableWrapper<V> implements Callable<V> {
+    Callable<V> callable;
```

Comment:
I think this field should be final, this can avoid multi-thread security issues.

Reply:
This field could be `final`, but I doubt it is thread security issue related. CallableWrapper is only created in parent thread, and didn't provide any way to update it. I can't see the thread issue.

Reply:
This problem occurs when someone maliciously manipulates the `callable` field in the same package.  so I suggest that the field should be `final`. The best way to resolve this problem is that this field be `final` and `private`

Reply:
`final Callable<V> callable;` is efficient . I'll change it